test/change.test.js

```diff
@@ -82,6 +82,36 @@ describe('Change', function() {
     });
   });
 
+  describe('Change.rectifyModelChanges - promise variant', function() {
+    describe('using an existing untracked model', function() {
+      beforeEach(function(done) {
+        var test = this;
+        Change.rectifyModelChanges(this.modelName, [this.modelId])
+          .then(function(trackedChanges) {
+            done();
+          })
+          .catch(function(err) {
+            return done(err);
+          });
```

Comment:
Hang on. Since these tests are checking whether the promise support works, we must be prepared to correctly handle the case when the support is broken, so that we get helpful test failures (assertion messages).

My example above would fail in a hard-to-debug way.

```js
beforeEach(function() {
  return Change.rectifyModelChanges(this.modelName, [this.modelId]);
});
```

When `Change.rectifyModelChanges` does not return a promise, Mocha will think the beforeEach hook was synchronous and will immediately invoke the next test case. That test case would start before the test environment was set up, it will fail and most likely with a confusing error message.

@jannyHou So you are right in using `function(done)` approach here, please keep it using in all tests you are adding as part of this patch.

However, please do simplify it as I shown above.

```js
beforeEach(function(done) {
  Change.rectifyModelChanges(this.modelName, [this.modelId])
    .then(function(trackedChanges) { done(); }, done);
});
```